### ISSUE TITLE
Support build time environment variables

### ISSUE THREAD

**Issue by that0n3guy** [dokku-alt/dokku-alt#207]
There are several things I'm forced to do in my dockerfile at "boot" rather than build... simply because I don't have environment variables at build time.

This looks like its being proposed: https://github.com/docker/docker/pull/9176

Is there a work around for now?  Maybe something that echo's the apps environment variables with `ENV` into the dockerfile right before build?  Maybe our own custom {{ENV}} string that gets replaced before build?  Is this a bad idea? :)

**Comment by ayufan** [comment]
It's always problem. I even though that it could be possible to inject ENVs during ONBUILD steps, but did not succeed.

I need to rethink that.

**Comment by that0n3guy** [comment]
How about this, right after you clone the repo to /tmp OR right before build... run a script like: https://gist.github.com/that0n3guy/e392d763df4b21c04fd3

It looks simply takes line by line takes the ~dokku/app/ENV file (one line looks like `export MYSQL_USER=whatever`) turns them into one long string like: `ENV MYSQL_USER=whateve FOO=bar MYSQL_DB=thinger`.  

Then it looks in `$DOCKERFILE` for the text `#dokkuEnv#`... if found, it replaces it with the string.

So a dockerfile that looks like: 

```
FROM that0n3guy/baseimage-nginx-phpfpm

#dokkuEnv#

ADD . /app/
RUN rm /app/Dockerfile && mv /app/.docker /build/

RUN chmod +x /build/.docker/*.sh && chmod +x /build/.docker/*/*.sh
RUN bash /build/.docker/addSSHkey.sh
RUN bash /build/.docker/gitCheckout.sh

RUN composer self-update

# Give web server permissions to access app folder
RUN chown -R www-data:www-data /app
```

becomes: 
```
FROM that0n3guy/baseimage-nginx-phpfpm

ENV MYSQL_USER=whateve FOO=bar MYSQL_DB=thinger

ADD . /app/
RUN rm /app/Dockerfile && mv /app/.docker /build/

RUN chmod +x /build/.docker/*.sh && chmod +x /build/.docker/*/*.sh
RUN bash /build/.docker/addSSHkey.sh
RUN bash /build/.docker/gitCheckout.sh

RUN composer self-update

# Give web server permissions to access app folder
RUN chown -R www-data:www-data /app
```

What do you think?

**Comment by ayufan** [comment]
That can work. I'm looking through docks and looking for the best approach.

**Comment by drbraden** [comment]
Is there a way to pass environment variables to the buildpack step?  I had an issue with the standard ruby buildpack timing-out during the fetch process on a slow VPS.  In fetcher.rb, the call to curl has timeout set to `ENV['CURL_TIMEOUT'] || 30` but I'm not sure how to set the env var.

It was easy enough to fork the buildpack and set a higher default, but I'd rather just pass a var if I can.

**Comment by ayufan** [comment]
Yes. Simply pass configs to app. They should get injected into app during build.

**Comment by that0n3guy** [comment]
@ayufan Any more thoughts on this?
